### DISCUSSION TITLE
optimize option for legacy images

### DISCUSSION THREAD

**Discussion by eswanzey** [vanjarosoftware/Vanjaro.Platform#1879] [Ideas]
The hope is to have a caution indicator when an image is conventionally placed (i.e., a plain old image containing just img and src references) within a block when editing a page.

And also have an option to act upon that image (e.g., "Optimize"), whereby the individual image, or all images on the page, would be optimized the Vanjaro way with the creation of alternates within the .versions subdirectory.

**Comment by meetmandeep** [comment]
> The hope is to have a caution indicator when an image is conventionally placed (i.e., a plain old image containing just img and src references) within a block when editing a page.

How are you adding the conventional code? Using Custom Code Block?

**Reply by eswanzey** [reply]
yes, mostly with Custom Code block. But also, when using native Vanjaro apps and inserting an image that was either ftp'd in or pre-existing to the Vanjaro platform install - they bypass the native Vanjaro optimizations too.

**Comment by eswanzey** [comment]
One thing to add on this. In the day to day editing of a hybrid site, the issue of orphan assets can be chipped away at without any performance penalty.

The Vanjaro logic seems to be that when placing images, "if a .resources asset exists" then create the code necessary to take advantage of it.

If the logic was, "if a .resources asset doesn't exist" then create the .resources assets and insert the necessary code - then we could target those high value orphan assets that are dispersed across a migrated site.
